motr/client.c

```diff
@@ -830,6 +830,66 @@ M0_INTERNAL int m0_op_init(struct m0_op *op,
 	return M0_RC(0);
 }
 
+M0_INTERNAL int m0_calculate_md5_inc_digest(
+		struct m0_md5_inc_digest_pi *pi,
+		struct m0_pi_seed *seed,
+		struct m0_bufvec *bvec,
+		enum m0_pi_calc_flag flag,
+		unsigned char *curr_digest,
+		unsigned char *checksum_without_seed)
+{
+	M0_ENTRY();
+
+	M0_PRE(pi != NULL);
+	M0_PRE(curr_digest != NULL);
+	M0_PRE(bvec != NULL);
+	M0_PRE(bvec->ov_vec.v_count != NULL);
+	M0_PRE(bvec->ov_buf != NULL);
+
+	/* Uncomment following code once motr has integration for MD5 */
+#if 0
+
+	/* This call is for first data unit, need to initialize prev_digest*/
+	if (flag & M0_PI_CALC_UNIT_ZERO) {
+		MD5_Init(&md5_digest_pi->prev_digest);
+	}
+
+	/* memcpy, so that we do not change the prev_digest */
+	memcpy(curr_digest, &md5_digest_pi->prev_digest, sizeof(MD5_CTX));
+
+	/* get the curr digest i by updating it*/
+	for (i = 0; i < bvec->ov_vec.v_nr; i++) {
+		MD5_Update(curr_digest, bvec->ov_buf[i], bvec->ov_vec.v_count[i]);
+	}
+
+	/* if caller want checksum without seed and with seed, in this case
+	 * 'checksum_without_seed' will be used to send wihout seed checksum
+	 */
+	if (checksum_without_seed) {
+		MD5_Final(checksum_without_seed, curr_digest);
```

Comment:
I believe appropriate error code from MD5_XXX will be taken into consideration

Reply:
Yes, this is not final code. I will add error checks and logs.